pkg/apis/controller/v1beta1/schedulingspec.go

```diff
@@ -69,3 +72,29 @@ type SchedulingSpecList struct {
 }
 
 type ResourceName string
+
+type ClusterReference struct {
+	Name string `json:"name"`
+}
+
+type ClusterSchedulingSpec struct {
+	Clusters        []ClusterReference    `json:"clusters,omitempty"`
+	ClusterSelector *metav1.LabelSelector `json:"clusterSelector,omitempty"`
+}
+
+type ScheduleTimeSpec struct {
+	Min     metav1.Time `json:"minTimestamp,omitempty"`
+	Desired metav1.Time `json:"desiredTimestamp,omitempty"`
+	Max     metav1.Time `json:"maxTimestamp,omitempty"`
+}
+
+type DispatchDurationSpec struct {
+	Expected int  `json:"expected,omitempty"`
+	Limit    int  `json:"limit,omitempty"`
+	Overrun  bool `json:"overrun,omitempty"`
+}
+
+type DispatchingWindowSpec struct {
```

Comment:
Is this something you're planning to implement later? I don't see it anywhere in the code.

Reply:
Yes, this spec could be used for sustainability, Diana has designed this spec.